src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -761,7 +761,23 @@ internal static TimeSpan TicksToTimeSpan(double ticks)
         private static Stream OpenStream(int fd, FileAccess access)
         {
             Debug.Assert(fd >= 0);
-            var socket = new Socket(new SafeSocketHandle((IntPtr)fd, ownsHandle: true));
+            var socketHandle = new SafeSocketHandle((IntPtr)fd, ownsHandle: true);
+            var socket = new Socket(socketHandle);
+
+            if (!socket.Connected)
+            {
+                // WSL1 workaround -- due to issues with sockets syscalls
+                // socket pairs fd's are erroneously inferred as not connected.
+                // Fall back to using FileStream instead.
+
+                GC.SuppressFinalize(socket);
+                GC.SuppressFinalize(socketHandle);
+
+                return new FileStream(
```

Comment:
A thing to note here is that the particular fallback will result in the `ReadAsync_OutputStreams_Cancel_RespondsQuickly` test to fail.